File: CareerPathfinder/CareerPathfinder.Core/Entities/Department.cs
using System.ComponentModel.DataAnnotations;

namespace CareerPathfinder.Core.Entities
{
    public class Department
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [MaxLength(500)]
        public string? Description { get; set; }

        public bool IsActive { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow; 

        public virtual ICollection<Question> Questions { get; set; } = new List<Question>();
        public virtual ICollection<Career> Careers { get; set; } = new List<Career>();
    }
}

File: src/types/department.ts
export interface Department {
  id: number;
  name: string;
  description?: string;
  questionCount: number;
  careerCount: number;
}

export interface DepartmentDto {
  id: number;
  name: string;
  description?: string;
  questionCount: number;
  careerCount: number;
}
